File: Algorithms/Implementation/Equalize the Array.php
<?php

/*
 * Complete the 'equalizeArray' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts INTEGER_ARRAY arr as parameter.
 */

function equalizeArray($arr) {
    # Write your code here
    $counts = array_count_values($arr);
    $max_count = max($counts);
    return count($arr) - $max_count;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$n = intval(trim(fgets(STDIN)));
$arr_temp = rtrim(fgets(STDIN));
$arr = array_map('intval', preg_split('/ /', $arr_temp, -1, PREG_SPLIT_NO_EMPTY));
$result = equalizeArray($arr);

fwrite($fptr, $result . "\n");
fclose($fptr);

File: Algorithms/Implementation/Equalize the Array.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

class Result
{

    /*
     * Complete the 'equalizeArray' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts INTEGER_ARRAY arr as parameter.
     */

    public static int equalizeArray(List<int> arr)
    {
        Dictionary<int, int> freqMap = new Dictionary<int, int>();

        foreach (int num in arr)
        {
            if (freqMap.ContainsKey(num))
            {
                freqMap[num]++;
            }
            else
            {
                freqMap[num] = 1;
            }
        }

        int maxFreq = freqMap.Max(kvp => kvp.Value);

        return arr.Count - maxFreq;
    }
}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int n = Convert.ToInt32(Console.ReadLine().Trim());

        List<int> arr = Console.ReadLine().TrimEnd().Split(' ').ToList().Select(arrTemp => Convert.ToInt32(arrTemp)).ToList();

        int result = Result.equalizeArray(arr);

        textWriter.WriteLine(result);

        textWriter.Flush();
        textWriter.Close();
    }
}
